test/model.translateAliases.test.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Test dependencies.
+ */
+
+var start = require('./common'),
+    assert = require('power-assert'),
+    mongoose = start.mongoose;
+
+describe('model translate aliases', function() {
+  it('should translate correctly', function() {
+    var Character = mongoose.model('Character', new mongoose.Schema({
+      name: { type: String, alias: '名' },
+      bio: {
+        age: { type: Number, alias: '年齢' }
+      }
+    }));
+
+    assert.deepEqual(
+      // Translate aliases
+      Character.translateAliases({
+        '名': 'Stark',
+        '年齢': 30
+      }),
+      // How translated aliases suppose to look like
+      {
```

Comment:
I'm not sure separating out dotted paths is the right behavior here. If you're querying by '年齢', you'd want to query `'bio.age': 30` rather than `bio: { age: 30 }`, because the latter will only match if `bio` is an object that contains exactly 1 key, `age`. On the other hand, `'bio.age': 30` would match even if `bio` has other keys, for example `birthday`.